app/gui/src/controller/graph/executed.rs

```diff
@@ -75,17 +75,17 @@ pub enum Notification {
 #[derive(Clone, CloneRef, Debug)]
 pub struct Handle {
     #[allow(missing_docs)]
-    pub logger:    Logger,
+    pub logger:        Logger,
     /// A handle to basic graph operations.
-    graph:         Rc<RefCell<controller::Graph>>,
+    graph:             Rc<RefCell<controller::Graph>>,
     /// Execution Context handle, its call stack top contains `graph`'s definition.
-    execution_ctx: model::ExecutionContext,
+    pub execution_ctx: model::ExecutionContext,
```

Comment:
This field was private for a reason: otherwise, someone could manually push a frame there without changing the `graph` field, which should always point to the definition on top of the frame.

Please, make it private and add `interrupt` and `restart` methods to this handle.